### PR TITLE
[mlir-lsp] Un-revert unit test additions

### PR DESCRIPTION
**Pull request by modocache**
This reverts the revert commit 6844c2feae93dd4251, which was comprised of the following commits:

1. f3f6f22dfcced - [mlir-lsp] Initialize `Reply::method` member (#89857)
2. 37e13d4924841 - [mlir-lsp] Log invalid notification params (#89856)
3. ba1b52e6e764a - [mlir-lsp] Add `outgoingNotification` unit test
4. 84bc21f910173 - [mlir-lsp] Add transport unit tests (#89855)

Of these, (4) specifically caused issues in Windows pre-merge buildbots, in the `TransportTest.MethodNotFound` unit test that it added. The failure was caused by a statement that asserted that opening a file stream on a newly created temporary file did not result in an error, but this assert failed on Windows.

This patch adds additional error logging for failures, to make it clearer what went wrong when failures occur. This patch also addresses the Windows failure by ensuring temporary files are created in the system temporary directory.

### CODE REVIEW

**Comment by modocache** [comment]
Note: I didn't add reviewers to this pull request for now; I'll wait until I get green pre-merge builds that confirm the fix is good.

**Comment by modocache** [comment]
The Windows pre-merge buildbot shows a successful run of the new tests.

**Review by walter-erquinigo — APPROVED** [review]

**Pull request merged**
